### PR TITLE
ci: execute all e2e tests on PR label with one GHA workflow

### PR DESCRIPTION
**Pull request by supershal**
**What problem does this PR solve?**:
- Simplifies conditions to execute e2e jobs
- One job to control execution of all the e2e tests on a PR label
- Ensure devkit image is available on dockerhub before running tests
- Make it easy to navigate all e2e tests for current PR in one place in GHA UI

**Which issue(s) does this PR fix?**:

* https://jira.d2iq.com/browse/D2IQ-NUMBER

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: .github/workflows/run-all-e2e-tests.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  workflow_dispatch:
+  pull_request:
+    types: [labeled, synchronize]
+
+permissions:
+  contents: read
+  id-token: write
+
+jobs:
+  set-global-conditions:
```
Comment (supershal) [inline .github/workflows/run-all-e2e-tests.yaml:13]: GHA workflows does not allow condition at workflow level that can be shared between all the jobs.
This hack allows setting complicated `if` condition on all the jobs and make it easy for dependent job to inherit condition.

File: .github/workflows/run-all-e2e-tests.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    runs-on: ubuntu-latest
+    outputs:
+      runs_e2e_tests: ${{ steps.set_runs_e2e_tests.outputs.RUNS_E2E_TESTS }}
+    if: |
+      github.event_name == 'pull_request' &&
+      (
+        (github.event.action == 'labeled' &&  github.event.label.name == 'runs-e2e-tests-temp') ||
+        (github.event.action == 'synchronize' && contains(github.event.pull_request.labels.*.name, 'runs-e2e-tests-temp'))
```
Comment (supershal) [inline .github/workflows/run-all-e2e-tests.yaml:21]: Once PR is ready to merge, 'runs-e2e-tests-temp' will be renamed to 'runs-e2e-tests'

**Comment by supershal** [comment]
sample runs: https://github.com/mesosphere/konvoy-image-builder/actions/runs/4319872699
 https://github.com/mesosphere/konvoy-image-builder/actions/runs/4326141584

File: .github/workflows/run-all-e2e-tests.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# Runs Azure tests when pull request opened, repopened or synchronized
+name: E2E Tests - Run all tests on PR
```
Comment (dkoshkin) [inline .github/workflows/run-all-e2e-tests.yaml:2]: How does this interact with https://github.com/mesosphere/konvoy-image-builder/pull/717, can we still trigger individual provider's tests?

Comment (supershal) [reply .github/workflows/run-all-e2e-tests.yaml:2]: with `workflow_dispatch` we can still trigger individual provider's test manually from GHA UI. However they might not reflect on the `Status check` list of the PR. 
I can port logic in https://github.com/mesosphere/konvoy-image-builder/pull/717 to this PR.

Comment (supershal) [reply .github/workflows/run-all-e2e-tests.yaml:2]: ported configuration for executing tests on a given provider

**Comment by supershal** [comment]
The latest version of the ansible galaxy collection `community.general` broke redhat subscription. https://github.com/ansible-collections/community.general/issues/6258
https://github.com/mesosphere/konvoy-image-builder/pull/714/commits/745d4a5e967c8b7d9584b99171b352298bfe0a7f
pins ansible collection versions

**Comment by supershal** [comment]
All test passed: https://github.com/mesosphere/konvoy-image-builder/actions/runs/4558766250

**Review by dkoshkin — APPROVED** [review]
This is great!
image

Another great benefit is if an individual test fails rerunning it is much quicker now since you don't need to rebuild the image.

**Pull request merged**